src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -147,6 +147,9 @@ class CoreWorkerDirectTaskSubmitter {
   /// if a remote raylet tells us to spill the task back to the local raylet.
   const ClientID local_raylet_id_;
 
+  /// The gcs client to communicate with gcs server.
+  std::shared_ptr<gcs::GcsClient> gcs_client_;
```

Comment:
Would be nice to also document where this is necessary, e.g. actor creation needs to go through gcs .etc